site/content/docs/02-template-syntax.md

```diff
@@ -0,0 +1,592 @@
+
+```html
+<input type=checkbox>
+```
+
+---
+
+Attribute values can contain JavaScript expressions.
+
+```html
+<a href="page/{p}">page {p}</a>
+```
+
+---
+
+Or they can *be* JavaScript expressions.
+
+```html
+<button disabled={!clickable}>...</button>
+```
+
+---
+
+An expression might include characters that would cause syntax highlighting to fail in regular HTML, in which case quoting the value is permitted. The quotes do not affect how the value is parsed:
+
+```html
+<button disabled="{number !== 42}">...</button>
+```
+
+---
+
+When the attribute name and value match (`name={name}`), they can be replaced with `{name}`.
+
+```html
+<!-- These are equivalent -->
+<button disabled={disabled}>...</button>
+<button {disabled}>...</button>
+```
+
+---
+
+*Spread attributes* allow many attributes or properties to be passed to an element or component at once.
+
+An element or component can have multiple spread attributes, interspersed with regular ones.
+
+```html
+<Widget {...things}/>
+```
+
+
+### Text expressions
+
+* `{expression}`
+
+---
+
+Text can also contain JavaScript expressions:
+
+```html
+<h1>Hello {name}!</h1>
+<p>{a} + {b} = {a + b}.</p>
+```
+
+
+### HTML expressions
+
+* `{@html expression}`
+
+---
+
+In a text expression, characters like `<` and `>` are escaped. With HTML expressions, they're not.
+
+> Svelte does not sanitize expressions before injecting HTML. If the data comes from an untrusted source, you must sanitize it, or you are exposing your users to an XSS vulnerability.
+
+```html
+<div class="blog-post">
+	<h1>{post.title}</h1>
+	{@html post.content}
+</div>
+```
+
+
+### If blocks
+
+* `{#if expression}...{/if}`
+* `{#if expression}...{:else if expression}...{/if}`
+* `{#if expression}...{:else}...{/if}`
+
+---
+
+Content that is conditionally rendered can be wrapped in an if block.
+
+```html
+{#if answer === 42}
+	<p>what was the question?</p>
+{/if}
+```
+
+---
+
+Additional conditions can be added with `{:else if expression}`, optionally ending in an `{:else}` clause.
+
+```html
+{#if porridge.temperature > 100}
+	<p>too hot!</p>
+{:else if 80 > porridge.temperature}
+	<p>too cold!</p>
+{:else}
+	<p>just right!</p>
+{/if}
+```
+
+
+### Each blocks
+
+* `{#each expression as name}...{/each}`
+* `{#each expression as name, index}...{/each}`
+* `{#each expression as name, index (key)}...{/each}`
+* `{#each expression as name}...{:else}...{/each}`
+
+---
+
+Iterating over lists of values can be done with an each block.
+
+```html
+<h1>Shopping list</h1>
+<ul>
+	{#each items as item}
+		<li>{item.name} x {item.qty}</li>
+	{/each}
+</ul>
+```
+
+---
+
+An each block can also specify an *index*, equivalent to the second argument in an `array.map(...)` callback:
+
+```html
+{#each items as item, i}
+	<li>{i + 1}: {item.name} x {item.qty}</li>
+{/each}
+```
+
+---
+
+If a *key* expression is provided, Svelte will diff the list when data changes, rather than adding or removing items at the end.
```

Comment:
Maybe be more explicit about the key being what's used to decide the identity of elements  in the list.

Reply:
👍